### PR TITLE
[DowngradePhp70] Add DowngradeDefineArrayConstantRector

### PR DESCRIPTION
**Pull request by samsonasik**
part of https://github.com/rectorphp/rector/issues/6079

### CODE REVIEW

File: rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            [
+                new Const_(
+                    new Name(
+                        $node->args[0]->value->value
+                    ),
+                    $node->args[1]->value
+                )
+            ]
+        );
+        return $parent->consts[0];
```
Comment (samsonasik) [inline rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php:82]: @TomasVotruba do you have idea to create `const` assign definition? The code above define constant but without `const ` start, eg:

```diff
-const ANIMALS = [
+ANIMALS = [
     'dog',
     'cat',
     'bird'
```

Comment (samsonasik) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: it works with string `const ` prefix on const name definition https://github.com/rectorphp/rector/pull/6270/commits/73001fa5d1657f8a43cc7c4b56871b926224f47e :tada:

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php (hunk 1 above)
Comment (samsonasik) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: that's not add "const " prefix definiton, that's solved with add "const " on name instantiation.

Comment (TomasVotruba) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: Sorry, wrong link: https://github.com/rectorphp/php-parser-nodes-docs#phpparsernodestmtclassconst

Comment (samsonasik) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: that's class constant, not a "const NAMECONST = [...]" that is not a class constant, the define is define constant that not a class constant.

Comment (TomasVotruba) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: Ah, sorry. This one then: https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Stmt/Const_.php

Comment (samsonasik) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: I tried that previously, see https://github.com/rectorphp/rector/pull/6270/commits/dc6934f3c04bc98107b97d8a0131adcc73d93940 , the solution is add string "const " during Const_ name instantiation

Comment (TomasVotruba) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: I've just pushed it. Feel free to re-use

Comment (samsonasik) [reply rules/DowngradePhp70/Rector/FuncCall/DowngradeDefineArrayConstantRector.php]: ok, implemented at https://github.com/rectorphp/rector/pull/6270/commits/89dcb1ea255d6ca47990857827dcc3d05cba9e0e

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**